api/src/main/java/org/eclipse/microprofile/config/spi/ConfigProviderResolver.java

```diff
@@ -36,6 +38,7 @@
  * @author <a href="mailto:rmannibucau@apache.org">Romain Manni-Bucau</a>
  * @author <a href="mailto:emijiang@uk.ibm.com">Emily Jiang</a>
  */
+@ServiceConsumer(value = ConfigProviderResolver.class, effective = "active")
```

Comment:
I assume this is also `active` for similar reasons as mentioned above. We should also document why we do active here.

Reply:
Exactly! Since there has been resistance to the SLM solution, this makes the feature usable during assembly.

Reply:
Whether this gets documented in the spec is up for debate. I think we could come up with standard language would simplify applying this to other MP specs if desired. It would largely be the same words for every spec except for the identity (which I'd suggest simply going with the bsn of the API for consistency and avoid confusion).

Reply:
Will this work, given that, at the moment, the service loading isn't done using the thread context classloader?

See line 110:
```
                instance = loadSpi(ConfigProviderResolver.class.getClassLoader());
```

Reply:
> Will this work, given that, at the moment, the service loading isn't done using the thread context classloader?

Yes. For example, you can use a fragment to attach the META-INF/service resource and add any necessary Import-Package/DynamicImport-Package statement to this bundle.

Apache Aries SPI Fly is another option.

Reply:
Yeap! The SML spec never limited exactly how the loading was done and so within the past two years the Apache Aries SPI Fly implementation was enhanced to support the two argument load(Class, ClassLoader) method. The heuristic is to use the passed ClassLoader as fallback if no providers are found. So no code change here is required.

Reply:
> SML

SLM :-)

Reply:
Also, a thread context class loader does not help here as trying to use it would introduce timing issues similar to using setInstance. You would have to arrange to set the TCCL properly before being the first one to cause the ServiceLoader to load the impl class.

To be fair, there are also timing issues with SPI Fly in that you have to ensure it is started (perhaps via start levels) before anyone causes the ServiceLoader call to be made.

Using a fragment is probably the safest as it will be attached at resolve time which is by definition before any code in the host bundle can execute.